std/algorithm/searching.d

```diff
@@ -3913,72 +3913,81 @@ Do nothing if there is no match.
 Params:
     pred = The predicate that determines whether elements from each respective
         range match. Defaults to equality $(D "a == b").
+*/
+template skipOver(alias pred = "a == b")
+{
+    /**
     r1 = The $(REF_ALTTEXT forward range, isForwardRange, std,range,primitives) to
         move forward.
     r2 = The $(REF_ALTTEXT input range, isInputRange, std,range,primitives)
         representing the initial segment of $(D r1) to skip over.
+    e = The element to match.
 
-Returns:
-true if the initial segment of $(D r1) matches $(D r2) or $(D pred) evaluates to true,
-and $(D r1) has been advanced to the point past this segment; otherwise false, and
-$(D r1) is left in its original position.
- */
-bool skipOver(R1, R2)(ref R1 r1, R2 r2)
-if (isForwardRange!R1 && isInputRange!R2
-    && is(typeof(r1.front == r2.front)))
-{
-    static if (is(typeof(r1[0 .. $] == r2) : bool)
-        && is(typeof(r2.length > r1.length) : bool)
-        && is(typeof(r1 = r1[r2.length .. $])))
+    Returns:
+    true if the initial segment of $(D r1) matches $(D r2) or $(D pred) evaluates to true,
+    and $(D r1) has been advanced to the point past this segment; otherwise false, and
+    $(D r1) is left in its original position.
+    */
+    bool skipOver(R1, R2)(ref R1 r1, R2 r2)
+    if (is(typeof(binaryFun!pred(r1.front, r2.front))) &&
+        isForwardRange!R1 &&
+        isInputRange!R2)
     {
-        if (r2.length > r1.length || r1[0 .. r2.length] != r2)
+        static if (is(typeof(pred) : string) && pred == "a == b"
```

Comment:
If alternatives are a lot of aggravationm, then fine. But seeing `template skipOver` is already a sign that something has gotten too complicated...